File: 3864-count-the-number-of-computer-unlocking-permutations/count-the-number-of-computer-unlocking-permutations.cs
public class Solution {
    public int CountPermutations(int[] complexity) {
        int n = complexity.Length;

        for (int i = 1; i < n; i++) {
            if (complexity[i] <= complexity[0]) {
                return 0;
            }
        }

        long ans = 1;
        int mod = 1000000007;
        for (int i = 1; i < n; i++) {
            ans = (ans * i) % mod;
        }
        
        return (int)ans;
    }
}

File: 3864-count-the-number-of-computer-unlocking-permutations/count-the-number-of-computer-unlocking-permutations.py
class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        MOD = 10**9 + 7
        n = len(complexity)
        fi = complexity[0]

        for c in complexity[1:]:
            if c <= fi:
                return 0

        ans = 1
        for i in range(2, n):
            ans *= i

        return ans % MOD
